x-pack/plugins/alerting/server/routes/create_rule.test.ts

```diff
@@ -238,11 +244,96 @@ describe('createRuleRoute', () => {
     });
   });
 
+  it('allows providing a custom id in non-default space but logs warning and returns warning header', async () => {
+    const expectedResult = {
+      ...createResult,
+      id: 'custom-id',
+    };
+    const licenseState = licenseStateMock.create();
+    const router = httpServiceMock.createRouter();
+
+    createRuleRoute(router, licenseState, logger);
+
+    const [config, handler] = router.post.mock.calls[0];
+
+    expect(config.path).toMatchInlineSnapshot(`"/api/alerting/rule/{id?}"`);
+
+    rulesClient.create.mockResolvedValueOnce({
+      ...mockedAlert,
+      id: 'custom-id',
+    });
+    rulesClient.getSpaceId.mockReturnValueOnce('another-space');
+
+    const [context, req, res] = mockHandlerArguments(
+      { rulesClient },
+      {
+        params: { id: 'custom-id' },
+        body: ruleToCreate,
+      },
+      ['ok']
+    );
+
+    expect(await handler(context, req, res)).toEqual({
+      body: expectedResult,
+      headers: {
+        warning: `199 kibana "Using the "id" path parameter to create rules in a custom space will lead to unexpected behavior in 8.0.0. Consult the docs for more details."`,
```

Comment:
Using the `id` path parameter to create rules in a custom space leads to unexpected behavior in 8.0.  Consult the docs for details.

Where "the docs" is a link.

Reply:
@gchaps This shows up in an HTTP response header, so it won't be a clickable link. It will look like: `Using the "id" path parameter to create rules in a custom space will lead to unexpected behavior in 8.0.0. Consult the Alerting API docs at https://www.elastic.co/guide/en/kibana/current/create-rule-api.html for more details.`

Is that ok?

Reply:
Yes.

Reply:
Done!